lang/cs/Org.Apache.REEF.Wake/Avro/ProtocolSerializer.cs

```diff
@@ -68,12 +70,30 @@ public sealed class ProtocolSerializer
             typeof(ProtocolSerializer).GetMethod("Register", BindingFlags.Instance | BindingFlags.NonPublic);
 
         /// <summary>
-        /// Register all of the protocol messages using reflection.
+        /// Construct an uninitialized protocol serializer.
+        /// </summary>
+        [Inject]
+        public ProtocolSerializer()
+        {
+        }
+
+        /// <summary>
+        /// Construct an initialized protocol serializer.
         /// </summary>
         /// <param name="assembly">The Assembly object which contains the namespace of the message classes.</param>
         /// <param name="messageNamespace">A string which contains the namespace the protocol messages.</param>
         public ProtocolSerializer(Assembly assembly, string messageNamespace)
         {
+            Initialize(assembly, messageNamespace);
+        }
+
+        /// <summary>
+        /// Register all of the protocol messages using reflection.
+        /// </summary>
+        /// <param name="assembly">The Assembly object which contains the namespace of the message classes.</param>
+        /// <param name="messageNamespace">A string which contains the namespace the protocol messages.</param>
+        public void Initialize(Assembly assembly, string messageNamespace)
```

Comment:
We have to roll back these changes. All initialization should happen in the constructor.

Reply:
Done